File: mylibrary/src/main/java/com/xiaolu/mylibrary/net/RxObserver.java
package com.xiaolu.mylibrary.net;

import io.reactivex.Observer;
import io.reactivex.annotations.NonNull;
import io.reactivex.disposables.Disposable;

/**
 * ================================================
 *
 * @ProjectName: MyLibraryTool
 * @Package: com.xiaolu.mylibrary.net
 * @ClassName: RxObserver
 * @Description: 自定义rxjavaObserver
 * @Author: xiaol
 * @CreateDate: 2020/11/3 16:49
 * @UpdateUser: xiaol
 * @UpdateDate: 2020/11/3 16:49
 * @UpdateRemark: 更新说明
 * @Version: 1.0
 * ================================================
 */
public abstract class RxObserver<T> implements Observer<T> {
    private RxManager rxApiManager;
    private String mKey;
    public RxObserver(String key) {
        this.mKey = key;
        rxApiManager = RxManager.getInstance();
    }

    @Override
    public void onSubscribe(@NonNull Disposable d) {
        rxApiManager.add(mKey, d);
    }

    @Override
    public void onNext(@NonNull T t) {
        onSuccess(t);
    }

    @Override
    public void onError(@NonNull Throwable e) {
        onErrors(e);
    }

    @Override
    public void onComplete() {

    }

    public abstract void onSuccess(@NonNull T t);

    public abstract void onErrors(@NonNull Throwable e);
}

File: mylibraryKotlin/src/main/java/com/xiaolu/mylibrarykotlin/net/RxObserver.kt
package com.xiaolu.mylibrarykotlin.net

import com.xiaolu.mylibrarykotlin.net.RxManager.Companion.instance
import io.reactivex.Observer
import io.reactivex.disposables.Disposable

/**
 * ================================================
 *
 * @ProjectName: MyLibraryTool
 * @Package: com.xiaolu.mylibrary.net
 * @ClassName: RxObserver
 * @Description: java类作用描述
 * @Author: xiaol
 * @CreateDate: 2020/11/3 16:49
 * @UpdateUser: xiaol
 * @UpdateDate: 2020/11/3 16:49
 * @UpdateRemark: 更新说明
 * @Version: 1.0
 * ================================================
 */
abstract class RxObserver<T>(private val mKey: String) : Observer<T> {
    private val rxApiManager: RxManager? = instance
    override fun onSubscribe(d: Disposable) {
        rxApiManager!!.add(mKey, d)
    }

    override fun onNext(t: T) {
        onSuccess(t)
    }

    override fun onError(e: Throwable) {
        onErrors(e)
    }

    override fun onComplete() {}
    abstract fun onSuccess(t: T)
    abstract fun onErrors(e: Throwable)

}